test/versioning-strategies/prerelease.ts

```diff
@@ -0,0 +1,280 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {describe, it} from 'mocha';
+
+import {expect} from 'chai';
+import {PrereleaseVersioningStrategy} from '../../src/versioning-strategies/prerelease';
+import {Version} from '../../src/version';
+
+describe('PrereleaseVersioningStrategyVersioningStrategy', () => {
+  describe('with breaking change', () => {
+    const commits = [
+      {
+        sha: 'sha1',
+        message: 'feat: some feature',
+        files: ['path1/file1.txt'],
+        type: 'feat',
+        scope: null,
+        bareMessage: 'some feature',
+        notes: [],
+        references: [],
+        breaking: false,
+      },
+      {
+        sha: 'sha2',
+        message: 'fix!: some bugfix',
+        files: ['path1/file1.rb'],
+        type: 'fix',
+        scope: null,
+        bareMessage: 'some bugfix',
+        notes: [{title: 'BREAKING CHANGE', text: 'some bugfix'}],
+        references: [],
+        breaking: true,
+      },
+      {
+        sha: 'sha3',
+        message: 'docs: some documentation',
+        files: ['path1/file1.java'],
+        type: 'docs',
+        scope: null,
+        bareMessage: 'some documentation',
+        notes: [],
+        references: [],
+        breaking: false,
+      },
+    ];
+    it('can bump a major', async () => {
```

Comment:
Can you add a test for a breaking change on a various pre-release aspects? For example:

- 1.0.0-beta01 => 1.0.0-beta02
- 2.0.0-beta01 => 2.0.0-beta02
- 1.0.1-beta01 => 2.0.0-beta02
- 1.1.0-beta01 => 2.0.0-beta01
- 1.1.1-beta01 => 2.0.0-beta01

Reply:
> * 1.0.1-beta01 => 2.0.0-beta02

This one should probably be `1.0.1-beta01 => 2.0.0-beta01`?

Reply:
Yup, sorry about that. Copy and paste error, I suspect.

Of course, there's actually more subtlety here - if the breaking change is only breaking with respect to a new feature introduce in a beta, then it can just be beta02. But we don't really have that much information, as far as I'm aware. This is why I'm expecting to keep looking at changes fairly carefully, manually, in order to check that release-please is doing the right thing :)